src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -90,45 +90,90 @@ internal static ProcessModule GetMainModule(Process targetProcess)
             return mainModule;
         }
 
+        // Cache of the current process' parentId 
+        private static int? s_currentParentProcessId;
+        private static readonly int s_currentProcessId = Process.GetCurrentProcess().Id;
+
         /// <summary>
-        /// Retrieve the parent process of a process.
-        /// 
-        /// Previously this code used WMI, but WMI is causing a CPU spike whenever the query gets called as it results in 
-        /// tzres.dll and tzres.mui.dll being loaded into every process to conver the time information to local format.
-        /// For perf reasons, we result to P/Invoke.
-        /// </summary>
-        ///
-        /// <param name="current">The process we want to find the
-        /// parent of</param>
-        internal static Process GetParentProcess(Process current)
-        {
+        /// Gets the parent process id for the specified process
+        /// </summary>        
+        /// <remarks>
+        /// Caching is used for the current process
+        /// </remarks>
+        /// <param name="processId">the id of the process whose parent id should be retreived</param>
+        /// <returns>0 on failure, else a parent process id. Note that the parent process id may have been recycled</returns>
+        internal static int GetParentProcessId(int processId)
+        {            
             int parentPid = 0;
+            // This is a common query (parent id for the current process)
+            // Use cached value if available
+            if (processId == s_currentProcessId && s_currentParentProcessId.HasValue)
```

Comment:
It is there to avoid creating a new process object every time the id of the current process is needed